src/neo4j_genai/components/entity_relation_extractor.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+
+import enum
+import json
+import logging
+from typing import Any
+
+from pydantic import BaseModel, ValidationError, validate_call
+
+from neo4j_genai.components.types import Neo4jGraph, TextChunk, TextChunks
+from neo4j_genai.exceptions import LLMGenerationError
+from neo4j_genai.generation.prompts import ERExtractionTemplate, PromptTemplate
+from neo4j_genai.llm import LLMInterface
+from neo4j_genai.pipeline.component import Component
+
+logger = logging.getLogger(__name__)
+
+
+class EntityRelationExtractor(Component):
+    async def run(self, chunks: TextChunks, **kwargs: Any) -> Neo4jGraph:
+        # for each chunk, returns a dict with entities and relations keys
+        return Neo4jGraph(nodes=[], relationships=[])
+
+
+class OnError(enum.Enum):
+    RAISE = "RAISE"
+    IGNORE = "CONTINUE"
+
+
+class LLMEntityRelationExtractor(EntityRelationExtractor):
+    def __init__(
+        self,
+        llm: LLMInterface,
+        prompt_template: ERExtractionTemplate | str = ERExtractionTemplate(),
```

Comment:
While I understand passing in a `prompt_template` allows for full flexibility, I'm wondering why not just let users pass in a `str` instead of `ERExtractionTemplate`. I'm wondering if this makes it overly complicated for a user.

Reply:
The input param is a `Union` between prompt template and string :) At the moment, if string is provided, the user is responsible for building the full prompt, including injecting parameters - which in practice means adding an extra (custom) component in the pipeline to add the chunk.

Reply:
I think we could have a `PromptBuilder` component at some point, so that the prompt can be tuned, but let's see how things evolve.